akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorStashSpec.scala

```diff
@@ -478,6 +487,29 @@ class EventSourcedBehaviorStashSpec extends ScalaTestWithActorTestKit(EventSourc
       c ! GetValue(stateProbe.ref)
       stateProbe.expectMessage(State(5, active = true))
     }
+
+    "fail when stash has reached limit" in {
+      val probe = TestProbe()
+      val behavior = EventSourcedBehavior[String, String, String](
+        PersistenceId("stash-is-ful"),
+        "",
+        commandHandler = {
+          case (_, _) ⇒
+            Effect.stash()
+        },
+        (state, _) ⇒ state
+      )
+
+      val c = spawn(behavior)
+      val limit = system.settings.config.getInt("akka.persistence.typed.stash-capacity")
+      (0 to limit).foreach { n ⇒
+        c ! s"cmd-$n"
+      }
+      EventFilter[StashOverflowException](occurrences = 1).intercept {
+        c ! s"cmd-${limit + 1}"
+        probe.expectTerminated(c, 10.seconds)
+      }
+    }
```

Comment:
I don't think this is how it is supposed to work? It also sends the last message to dead letters and logs a bunch of things:

```
[WARN] [02/07/2019 17:34:17.980] [EventSourcedBehaviorStashSpec-akka.actor.default-dispatcher-2] [akka://EventSourcedBehaviorStashSpec/user/$a] received dead letter without sender: cmd-501
[WARN] [02/07/2019 17:34:17.980] [EventSourcedBehaviorStashSpec-akka.actor.default-dispatcher-2] [akka://EventSourcedBehaviorStashSpec/user/$a] received dead letter from Actor[akka://EventSourcedBehaviorStashSpec/system/recoveryPermitter#713166041]: RecoveryPermitGranted
[INFO] [02/07/2019 17:34:17.980] [EventSourcedBehaviorStashSpec-akka.actor.default-dispatcher-3] [akka://EventSourcedBehaviorStashSpec/user/$a] Message [java.lang.String] without sender to Actor[akka://EventSourcedBehaviorStashSpec/user/$a#-1582112716] was not delivered. [1] dead letters encountered. If this is not an expected behavior, then [Actor[akka://EventSourcedBehaviorStashSpec/user/$a#-1582112716]] may have terminated unexpectedly, This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
[INFO] [02/07/2019 17:34:17.980] [EventSourcedBehaviorStashSpec-akka.actor.default-dispatcher-3] [akka://EventSourcedBehaviorStashSpec/user/$a] Message [akka.persistence.RecoveryPermitter$RecoveryPermitGranted$] from Actor[akka://EventSourcedBehaviorStashSpec/system/recoveryPermitter#713166041] to Actor[akka://EventSourcedBehaviorStashSpec/user/$a#-1582112716] was not delivered. [2] dead letters encountered. If this is not an expected behavior, then [Actor[akka://EventSourcedBehaviorStashSpec/user/$a#-1582112716]] may have terminated unexpectedly, This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
```

Reply:
Good to test this.

How to handle overflow is defined with
```
akka.persistence.typed {
  internal-stash-overflow-strategy = akka.persistence.ThrowExceptionConfigurator
}
```

It might be better to use `DiscardConfigurator` as the default here?

We should probably have tests for both overflow configurations.

In this case I don't think it's even reaching your command handler. It's stashing in the `RequestingRecoveryPermit` before it got the permit to start. To test the stashing from the command handler you need to have a request-response roundtrip first before sending all the commands to stash. Both tests might be of value.

Reply:
I changed that to drop by default, additionally I got rid of the dynamic loading of a class when it wasn't really user pluggable and there was three concrete types out of which one isn't useable for typed persistence (respond to sender)

Didn't figure out a way to make a deterministic test of overflow while recovering etc, but since that is using the same logic this _should_ cover that as well.

Reply:
Looks good